src/js/controller/settings/exportimage/CExportController.js

```diff
@@ -0,0 +1,91 @@
+(function () {
+  var ns = $.namespace('pskl.controller.settings.exportimage');
+
+  var BLACK = '#000000';
+
+  ns.CExportController = function (piskelController) {
+    this.piskelController = piskelController;
+  };
+
+  pskl.utils.inherit(ns.CExportController, pskl.controller.settings.AbstractSettingController);
+
+  ns.CExportController.prototype.init = function () {
+
+    var downloadButton = document.querySelector('.c-download-button');
+    this.addEventListener(downloadButton, 'click', this.onCDownloadButtonClick_);
+  };
+
+  ns.CExportController.prototype.onCDownloadButtonClick_ = function (evt) {
+    var fileName = this.getPiskelName_() + '.c';
+    var cName = this.getPiskelName_().replace(' ','_');
+    var outputCanvas = this.getFramesheetAsCanvas();
+    var width = this.piskelController.getWidth();
+    var height = this.piskelController.getHeight();
+    var frameCount = this.piskelController.getFrameCount();
+
+    // Create a background canvas that will be filled with the transparent color before each render.
+    var background = pskl.utils.CanvasUtils.createCanvas(width, height);
+    var context = background.getContext('2d');
+    context.fillStyle = BLACK;
+
+    // Useful defines for C routines
+    var frameStr = '#include <stdint.h>\n\n';
+    frameStr += '#define ' + cName.toUpperCase() + '_FRAME_COUNT ' +  this.piskelController.getFrameCount() + '\n';
+    frameStr += '#define ' + cName.toUpperCase() + '_FRAME_WIDTH ' + width + '\n';
+    frameStr += '#define ' + cName.toUpperCase() + '_FRAME_HEIGHT ' + height + '\n\n';
+
+    frameStr += '/* Piskel data for \"' + this.getPiskelName_() + '\" */\n\n';
+
+    frameStr += 'static const uint32_t ' + cName.toLowerCase();
+    frameStr += '_data[' + frameCount + '][' + width * height + '] = {\n';
+
+    for (var i = 0 ; i < frameCount ; i++) {
+      var render = this.piskelController.renderFrameAt(i, true);
+      context.clearRect(0, 0, width, height);
```

Comment:
After reading this again, I think we can simplify it further: this.piskelController.renderFrameAt(i, true); actually returns a Canvas2d element, so you can directly call getContext('2d') to get the context etc ... No need to redraw it to a temporary canvas element! Sorry, lack of API documentation :/

[For information, if we kept the current approach we would still need at least the clearRect. The code is drawing every frame in the same canvas. Before drawing frame N+1 you need to cleanup the canvas from frame N. If frame N+1 has transparent pixels, they won't override the pixels from frame N.]